mlog/logr.go

```diff
@@ -39,30 +39,32 @@ type LogTargetCfg map[string]*LogTarget
 type LogrCleanup func() error
 
 func newLogr(targets LogTargetCfg) (*logr.Logger, error) {
-	var errs error
-
-	lgr := logr.Logr{}
-
+	lgr := &logr.Logr{}
 	lgr.OnExit = func(int) {}
 	lgr.OnPanic = func(interface{}) {}
-
 	lgr.OnLoggerError = onLoggerError
 	lgr.OnQueueFull = onQueueFull
 	lgr.OnTargetQueueFull = onTargetQueueFull
 
+	err := logrAddTargets(lgr, targets)
+	logger := lgr.NewLogger()
+	return &logger, err
+}
+
+func logrAddTargets(lgr *logr.Logr, targets LogTargetCfg) error {
+	var errs error
 	for name, t := range targets {
-		target, err := newLogrTarget(name, t)
+		target, err := NewLogrTarget(name, t)
 		if err != nil {
 			errs = multierror.Append(err)
 			continue
 		}
 		lgr.AddTarget(target)
 	}
-	logger := lgr.NewLogger()
-	return &logger, errs
+	return errs
 }
 
-func newLogrTarget(name string, t *LogTarget) (logr.Target, error) {
+func NewLogrTarget(name string, t *LogTarget) (logr.Target, error) {
```

Comment:
Since we export this I'd probably add a godoc line explaining what it does.

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/15076/commits/04e503a93b678005c9c71ebd5ad897bede654ce1